lib/generator.js

```diff
@@ -406,7 +406,7 @@ class Generator {
       const targetFile = path.resolve(this.targetDir, relativeBaseDir, newFileName);
       const relativeTargetFile = path.relative(this.targetDir, targetFile);
 
-      const shouldOverwriteFile = this.shouldOverwriteFile(relativeTargetFile);
+      const shouldOverwriteFile = await this.shouldOverwriteFile(relativeTargetFile);
```

Comment:
To simplify we could do:
```js
if (!this.shouldOverwriteFile(relativeTargetFile)) return;
```
since the variable are never used.

Reply:
you also need to await. But, await with and NOT in a if check doesn't look nice.

Reply:
@jonaslagoni suggestion is also nice. But I'm fine with this fix also.

Note: should fix the other call first.

Reply:
Yea you are right @WaleedAshraf since the function is `async` it should use `await` even though it only returns a boolean otherwise it will skip the step completely.